src/pyggdrasil/analyze/_calculation.py

```diff
@@ -0,0 +1,92 @@
+"""Analyze trees from MCMC runs."""
+
+import jax.numpy as jnp
+from typing import Union, Callable
+import logging
+
+from pyggdrasil import TreeNode, compare_trees
+from pyggdrasil.tree_inference import Tree
+
+from pyggdrasil.interface import PureMcmcData
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+def analyze_mcmc_run(
+    mcmc_data: PureMcmcData,
+    metric: Callable[[TreeNode, TreeNode], float],
+    base_tree: TreeNode,
+) -> tuple[list[int], list[float]]:
+    """Analyze a MCMC run.
+
+    Args:
+        mcmc_data : PureMcmcData
+                    MCMC run data to analyze of iteration no.,
+                    tree, and log-probability
+        metric :  Callable[[TreeNode, TreeNode], float]
+                     metric to apply to the trees.
+        base_tree : TreeNode
+                    Tree to compare all applicable metrics to.
+
+    Returns:
+        list[int], list[float]
+        Iteration number and results of the metric.
+    """
```

Comment:
This is one function that allows calculating all [TreeNode, TreeNode] -> float metrics, that is all distances and also `is_true_tree`

Simples design I could think of.